### PR TITLE
Fix #25: allow using repeat(1) marker to disable repeat

### LINKED ISSUE #21: Does not work with parametrized nodeids
`pytest tests/test_database.py::TestDatabaseFixtures::test_access\[db\]` works, but adding `--count 10` fails:

> collecting 0 items                                                                                                                                                                           
> ERROR: not found: …/Vcs/pytest-django/tests/test_database.py::TestDatabaseFixtures::test_access[db]                                                                               
> (no name '…/Vcs/pytest-django/tests/test_database.py::TestDatabaseFixtures::test_access[db]' in any of [<Instance '()'>])                                   

pytest version 3.6.3.dev5+gb7b9c54d

### LINKED ISSUE #25: Not possible to disable repetitions for a certain test case
It is not possible to disable repetitions for a certain test case. It fails with a `TypeError` exception if the repeat mark is set to 1. It is possible to limit it to 2, but 1 causes a failure. Here is an example:

```python
@pytest.mark.repeat(1)
def test_heavy():
    pass

def test_light():
    pass
```

`py.test --count=10` fails with:
> TypeError: issubclass() arg 1 must be a class

### PR DESCRIPTION
**Pull request by vbarbaresi**
For instance:

```python
@pytest.mark.repeat(1)
def test_heavy():
    pass
```

When ran with  `py.test --count 10`, this doesn't crash anymore
and runs the test once.

### CODE REVIEW

File: pytest_repeat.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ class UnexpectedError(Exception):
 
 @pytest.fixture(autouse=True)
 def __pytest_repeat_step_number(request):
-    if request.config.option.count > 1:
+    marker = request.keywords.get('repeat')
+    count = marker and marker.args[0] or request.config.option.count
+    if count > 1:
```
Comment (blueyed) [inline pytest_repeat.py]: Should this maybe only be used if there is a config option / arg also/already?  (i.e. still have the outer `if request.config.option.count > 1:`)

Or is it a valid use case to repeat some tests always?

File: pytest_repeat.py [hunk 2]
```diff
@@ -36,7 +36,9 @@ class UnexpectedError(Exception):
 
 @pytest.fixture(autouse=True)
 def __pytest_repeat_step_number(request):
-    if request.config.option.count > 1:
+    marker = request.keywords.get('repeat')
```
[the remark this answers is not in the document]
Comment (vbarbaresi) [reply pytest_repeat.py]: Which code below did you mean?

I would like to use `marker = request.node.get_closest_marker("repeat")` 
But to preserve compatibility with pytest 2.8, I have to use `request.node.get_marker("repeat")` which triggers a deprecation warning on the latest pytest. So I'm a little lost here

[the remark this answers is not in the document]
Comment (vbarbaresi) [reply pytest_repeat.py]: Easy now that pytest >= 3.6 is required ✅

**Comment by gogobera** [comment]
Is there a reason this hasn't been/can't be merged? Merge conflicts must be minimal.

**Comment by nicoddemus** [comment]
> Is there a reason this hasn't been/can't be merged? Merge conflicts must be minimal.

None, most likely it slipped people's minds. I've fixed the conflict and will merge once it passes, thanks for the ping!

**Comment by vbarbaresi** [comment]
thanks!

**Comment by nicoddemus** [comment]
@vbarbaresi thanks again for the contribution and really sorry this got stuck for so long, we really did drop the ball here. 😕

**Pull request merged**